packages/api/cms-api/src/user-permissions/user.resolver.ts

```diff
@@ -20,10 +17,21 @@ export class UserResolver {
         return this.userService.getUser(id);
     }
 
-    @Query(() => PaginatedUserList)
-    async userPermissionsUsers(@Args() args: FindUsersArgs): Promise<PaginatedUserList> {
+    @Query(() => UserPermissionsPaginatedUserList)
+    async userPermissionsUsers(@Args() args: FindUsersArgs): Promise<UserPermissionsPaginatedUserList> {
         const [users, totalCount] = await this.userService.findUsers(args);
-        return new PaginatedUserList(users, totalCount, args);
+        return new UserPermissionsPaginatedUserList(
+            await Promise.all(
+                users.map(async (user) => {
+                    return {
+                        ...user,
+                        permissionsCount: (await this.userService.getPermissions(user)).length,
+                        contentScopesCount: (await this.userService.getContentScopes(user)).length,
+                    };
```

Comment:
If you convert those to a FieldResolver, you would not need a new Type. The list query (userPermissionsUsers) and detail query (userPermissionsUserById) should always return the same type.